### ISSUE TITLE
Logs: Provide option to toggle \n and \t characters in log messages to literal newlines and tabs

### ISSUE THREAD

**Issue by ethirolle** [grafana/grafana#27964]
Per https://github.com/grafana/loki/issues/2562#issuecomment-700237334, I am moving this issue from Loki repo to Grafana repo:

**Is your feature request related to a problem? Please describe.**
Many logs (e..g., from java or SQL) will include log messages that embed a lot of `\n` or `\r\n` or `\t` characters. These formatted log messages are pretty hard to read when viewed in Loki. They would be much easier to read if we could toggle into a view where these formatting characters are rendered as actual newlines & tabs.

Example screenshot:
![image](https://user-images.githubusercontent.com/10763196/91461143-a453ac80-e856-11ea-8656-d75ef6b348a4.png)

**Describe the solution you'd like**
The Loki UI depends somewhat on a "one log entry per line" view. When you click on a log line, you get a pop-out with the log labels and parsed fields.

My idea is to provide a tab or toggle in the Loki log UI that allows the user to toggle into a view where the log message is displayed with literal newlines and tabs substituted for the `\n` and `\t` characters.

**Describe alternatives you've considered**
I have read through https://github.com/grafana/loki/issues/74, which discusses a related-but-different problem: i.e., that is basically the inverse problem: how to parse logging in such a way that a multi-line entry (like a stack trace) is treated as a single log entry.

I do not see any other existing GH issues that treat this exact same feature request.

**Additional context**
Here is an example of one log line pulled from Loki, which shows the kind of formatting that is typical in some logging:
[one-log-line_prettified.JSON.txt](https://github.com/grafana/grafana/files/5311958/one-log-line_prettified.JSON.txt)

and here is how that **message** looks like after substituting literal newlines for every "\n":
[one-log-line-MESSAGE-with-newlines.txt](https://github.com/grafana/grafana/files/5311964/one-log-line-MESSAGE-with-newlines.txt)

**Comment by aocenas** [comment]
Also related to this a bit is https://github.com/grafana/grafana/issues/18522 which is about formatting json. That would not help here cause this is about nested field that is separately formatted/encoded but probably should be looked at together.

**Comment by ivanahuckova** [comment]
But this should work. I have taken example log line and using "Wrap log line" fixed the view according to what you expected. 🤔 Can I ask what Grafana version you are using? 
![image](https://user-images.githubusercontent.com/30407135/97612796-28bdd980-1a18-11eb-87a8-fb30bad4df05.png)

**Comment by ivanahuckova** [comment]
@ethirolle could you maybe use **Query Inspector** and share what does the response and log line exactly look like when Grafana receives it. Because I have taken your _one-log-line_prettified.JSON.txt_ and used it as a loki log message and using **wrap log lines** I got correct formatting. So I am just thinking if there couldn't be some escaping error or something. 🤔
![image](https://user-images.githubusercontent.com/30407135/97615012-07122180-1a1b-11eb-9f69-b5a5e3cf2984.png)

**Comment by ivanahuckova** [comment]
This feature would be helpful for the users, who have incorrectly escaped logs/parts of the logs (e.g. sql text, tracebacks), but it is hard for them to escape it correctly. We could improve their experience by adding a toggle/button/options that would for every place there is a "\n", replace that with an actual newline and same for "\t" (replace with an actual tab).

**Comment by aocenas** [comment]
@ivanahuckova btw how do we display it in the log details? Maybe we could do some heuristic there and unescape `message` field automatically?

**Comment by sourcehawk** [comment]
This issue still persists. It was only implemented as a feature in the explore tab

**Comment by Tensai123** [comment]
Is it available on panel in dashboard?

**Comment by MedAzizTousli** [comment]
Are we going to have this feature in Logs visualization or not?

**Comment by ifrost** [comment]
cc @grafana/observability-logs

**Comment by matyax** [comment]
Yes, we can have this in the new visualization. Adding it there: https://github.com/grafana/grafana/issues/99075?r

**Comment by matyax** [comment]
Done in https://github.com/grafana/grafana/issues/99075?
